jdaviz/tests/test_app.py

```diff
@@ -17,3 +17,58 @@ def test_get_tray_item_from_name():
 
     with pytest.raises(KeyError, match='not found in app'):
         app.get_tray_item_from_name('imviz-compass')
+
+
+def test_duplicate_data_labels(specviz_helper, spectrum1d):
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    dc = specviz_helper.app.data_collection
+    assert dc[0].label == "test"
+    assert dc[1].label == "test (1)"
+    specviz_helper.load_spectrum(spectrum1d, data_label="test_1")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    assert dc[2].label == "test_1"
+    assert dc[3].label == "test (2)"
+
+
+def test_duplicate_data_labels_with_brackets(specviz_helper, spectrum1d):
+    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
+    dc = specviz_helper.app.data_collection
+    assert len(dc) == 2
+
+
+def test_return_data_label_is_none(specviz_helper):
+    data_label = specviz_helper.app.return_data_label(None)
+    assert data_label == "Unknown"
+
+
+def test_return_data_label_is_image(specviz_helper):
+    data_label = specviz_helper.app.return_data_label("data/path/test.jpg")
+    assert data_label == "test[jpg]"
+
+
+def test_hdulist_with_filename(cubeviz_helper, image_cube_hdu_obj):
+    image_cube_hdu_obj.file_name = "test"
+    data_label = cubeviz_helper.app.return_data_label(image_cube_hdu_obj)
+    assert data_label == "test[HDU object]"

```

Comment:
Does this logic route have a real use case? When I pass in HDUList, I expect one of the following (mutually exclusive):

1. Each individual HDU has its own label (as per Cubeviz default behavior).
2. The first valid HDU is loaded and has a label just for that one (as per Imviz default behavior).

I am not sure what this is supposed to do. None of the viz loads the actual HDUList as a whole (if you understand what I mean).

Reply:
There's this: https://github.com/spacetelescope/jdaviz/blob/main/jdaviz/configs/imviz/plugins/parsers.py#L68
But then there are so many private methods in the imviz parser it's hard to track down what happens with HDUList objects there.

Also this in cubeviz: https://github.com/spacetelescope/jdaviz/blob/c03cd50845a3728bce87a4d56e202b6e349110fa/jdaviz/configs/cubeviz/plugins/parsers.py#L128-L132

Reply:
Are we sure that is the case currently and will stay the case in the future? Do you suggest I remove that check?

Reply:
Other devs can weight in but I think you can remove this check.